Indicators/RelativeVigorIndex.cs

```diff
@@ -0,0 +1,210 @@
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => CloseBand.IsReady && RangeBand.IsReady;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+        
+        private readonly RollingWindow<IBaseDataBar> _previousInputs;
+
+        /// <summary>
+        /// Gets the band of Closes for the RVI.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> CloseBand { get; }
+
+        /// <summary>
+        /// Gets the band of Ranges for the RVI.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> RangeBand { get; }
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class.
+        /// </summary>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(int period, MovingAverageType type)
+            : this($"RVI({period},{type})", period, type)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class.
+        /// </summary>
+        /// <param name="name">The name of this indicator.</param>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(string name, int period, MovingAverageType type = MovingAverageType.Simple)
+            : base(name)
+        {
+            WarmUpPeriod = period + 3;
+            CloseBand = type.AsIndicator("_closingBand", period);
+            RangeBand = type.AsIndicator("_rangeBand", period);
+            _previousInputs = new RollingWindow<IBaseDataBar>(3);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class along with a signal term.
+        /// </summary>
+        /// <param name="name">The name of this indicator.</param>
+        /// <param name="signalName">The name of the signal associated with this indicator.</param>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(
+            string name, string signalName, int period, MovingAverageType type = MovingAverageType.Simple)
+            : base(name)
+        {
+            WarmUpPeriod = period + 3;
+            CloseBand = type.AsIndicator("_closingBand", period);
+            RangeBand = type.AsIndicator("_rangeBand", period);
+            _previousInputs = new RollingWindow<IBaseDataBar>(3);
+            Signal = new RviSignal(signalName);
```

Comment:
Wondering if we should directly no make it optional and have it there always, simplify the logic, not sure it's worth not having it (null checks, duplicated constructors, etc). Maybe can take the indicator name and directly add `_S` to it

Reply:
`RelativeVigorIndexSignal` & agree should be there all the time like MACD.

Reply:
Will do -- I reckon it does not add much in terms of workload, anyways